### PR TITLE
Fixed unit test, remove BPT and integrate it to the interface.

### PR DESCRIPTION
**Pull request by tinkler**
fix: Temporarily removed type judgment due to version-dependent issues
perf: remove unused StakedToken
feat: deploy a StakedTokenDataProvider which work
with interface; remove BPT

### CODE REVIEW

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 1]
```diff
@@ -1,147 +1,105 @@
...
+    address stakedAsset,
+    address oracle
+  ) external view returns (StakedTokenData memory stakedAssetData);
 
   /**
-   * @notice Return the address of the Staked PSYS token
-   * @return The address of the StkPSYS token
+   * @notice Retrieves staked token data and user-specific data for a given user, staked asset,
+   *          and its associated oracle.
+   * @param stakedAsset The address of the staked asset (eg. stkAave, stkBptAave)
```
Comment (Potolski) [inline contracts/interfaces/IStakedTokenDataProvider.sol]:
```suggestion
   * @param stakedAsset The address of the staked asset (eg. stkPSYS)
```

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 2]
```diff
@@ -1,147 +1,105 @@
...
+   * @return The address of the StkAAVE token
    */
-  function BPT_PRICE_FEED() external returns (address);
+  function STAKED_PSYS() external view returns (address);
 
   /**
-   * @notice Return the address of the PSYS token
-   * @return The address of the PSYS token
+   * @notice Returns data of Staked asset
+   * @param stakedAsset The address of the staked asset (eg. stkAave, stkBptAave)
```
Comment (Potolski) [inline contracts/interfaces/IStakedTokenDataProvider.sol]:
```suggestion
   * @param stakedAsset The address of the staked asset (eg. stkPSYS)
```

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 3]
```diff
@@ -1,147 +1,105 @@
...
-   * @return The address of the PSYS price feed, ETH denominated, expressed with 18 decimals
+   * @return The address of the AAVE price feed, USD denominated, expressed with 8 decimals
    */
-  function PSYS_PRICE_FEED() external returns (address);
+  function PSYS_USD_PRICE_FEED() external view returns (address);
 
   /**
-   * @notice Return the address of the BPT price feed, ETH denominated
-   * @return The address of the BPT price feed, ETH denominated, expressed with 18 decimals
+   * @return The address of the StkAAVE token
```
Comment (Potolski) [inline contracts/interfaces/IStakedTokenDataProvider.sol]:
```suggestion
   * @return The address of the StkPSYS token
```

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 4]
```diff
@@ -1,147 +1,105 @@
...
-   * @notice Return the address of the ETH price feed, USD denominated
    * @return The address of the ETH price feed, USD denominated, expressed with 8 decimals
    */
-  function ETH_USD_PRICE_FEED() external returns (address);
+  function ETH_USD_PRICE_FEED() external view returns (address);
 
   /**
-   * @notice Return the address of the PSYS price feed, ETH denominated
-   * @return The address of the PSYS price feed, ETH denominated, expressed with 18 decimals
+   * @return The address of the AAVE price feed, USD denominated, expressed with 8 decimals
```
Comment (Potolski) [inline contracts/interfaces/IStakedTokenDataProvider.sol]:
```suggestion
   * @return The address of the PSYS price feed, USD denominated, expressed with 18 decimals
```

File: tasks/deployments/deploy-StakedTokenDataProvider.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { deployStakedTokenDataProvider } from '../../helpers/contracts-accessors';
+import { checkVerification } from '../../helpers/etherscan-verification';
+import { registerContractInJsonDb } from '../../helpers/contracts-helpers';
+
+const { StakedTokenDataProvider, StakedPSYSV3 } = eContractid;
+
+task(`deploy-${StakedTokenDataProvider}`, `Deploys the ${StakedTokenDataProvider} contract`)
+  .addParam('stkAave', `The address of the ${StakedPSYSV3} contract`)
+  .addParam('ethOracle', 'The address of the eth price feed contract')
+  .addParam('psysOracle', 'The address of the aave price feed contract')
```
Comment (Potolski) [inline tasks/deployments/deploy-StakedTokenDataProvider.ts]:
```suggestion
  .addParam('psysOracle', 'The address of the psys price feed contract')
```

File: tasks/deployments/deploy-StakedTokenDataProvider.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { getAddress } from 'ethers/lib/utils';
+import { ZERO_ADDRESS } from '../../helpers/constants';
+import { deployStakedTokenDataProvider } from '../../helpers/contracts-accessors';
+import { checkVerification } from '../../helpers/etherscan-verification';
+import { registerContractInJsonDb } from '../../helpers/contracts-helpers';
+
+const { StakedTokenDataProvider, StakedPSYSV3 } = eContractid;
+
+task(`deploy-${StakedTokenDataProvider}`, `Deploys the ${StakedTokenDataProvider} contract`)
+  .addParam('stkAave', `The address of the ${StakedPSYSV3} contract`)
```
Comment (Potolski) [inline tasks/deployments/deploy-StakedTokenDataProvider.ts]:
```suggestion
  .addParam('stkPSYS', `The address of the ${StakedPSYSV3} contract`)
```

File: tasks/deployments/deploy-StakedTokenDataProvider.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  .addParam('ethOracle', 'The address of the eth price feed contract')
+  .addParam('psysOracle', 'The address of the aave price feed contract')
+  .addFlag('verify', 'Verify StakedPSYSV3 contract via Etherscan API.')
+  .setAction(
+    async (
+      {
+        stkAave: stkAaveAddress,
```
Comment (Potolski) [inline tasks/deployments/deploy-StakedTokenDataProvider.ts]:
```suggestion
        stkPSYS: stkPSYSAddress,
```

File: tasks/deployments/deploy-StakedTokenDataProvider.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        psysOracle: psysOracleAddress,
+        verify,
+      },
+      localBRE
+    ) => {
+      await localBRE.run('set-dre');
+      if (verify) {
+        checkVerification();
+      }
+      if (!stkAaveAddress || getAddress(stkAaveAddress) === ZERO_ADDRESS) {
```
Comment (Potolski) [inline tasks/deployments/deploy-StakedTokenDataProvider.ts]:
```suggestion
      if (!stkPSYSAddress || getAddress(stkPSYSAddress) === ZERO_ADDRESS) {
```

File: tasks/deployments/deploy-StakedTokenDataProvider.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        throw `address of stkAave contract is invalid`;
```
Comment (Potolski) [inline tasks/deployments/deploy-StakedTokenDataProvider.ts]:
```suggestion
        throw `address of stkPSYS contract is invalid`;
```

File: tasks/deployments/deploy-StakedTokenDataProvider.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      }
+      if (!ethOracleAddress || getAddress(ethOracleAddress) === ZERO_ADDRESS) {
+        throw `address of ethOracle contract is invalid`;
+      }
+      if (!psysOracleAddress || getAddress(psysOracleAddress) === ZERO_ADDRESS) {
+        throw `address of psysOracle contract is invalid`;
+      }
+
+      const stakedTokenDataProvider = await deployStakedTokenDataProvider(
+        [stkAaveAddress, ethOracleAddress, psysOracleAddress],
```
Comment (Potolski) [inline tasks/deployments/deploy-StakedTokenDataProvider.ts]:
```suggestion
        [stkPSYSAddress, ethOracleAddress, psysOracleAddress],
```

File: contracts/misc/StakedTokenDataProvider.sol [hunk 11]
```diff
@@ -1,185 +1,159 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.7.5;
+pragma solidity ^0.7.5;
 pragma experimental ABIEncoderV2;
 
 import {IERC20} from '../interfaces/IERC20.sol';
-import {AggregatedStakedPSYSV3} from '../interfaces/AggregatedStakedPSYSV3.sol';
```
Comment (Potolski) [inline contracts/misc/StakedTokenDataProvider.sol]: why did you change this up for IAggregatedStakeToken?

File: package.json [hunk 12]
```diff
@@ -14,7 +14,7 @@
     "compile": "SKIP_LOAD=true hardhat compile",
     "compile:force": "npm run compile -- --force",
     "compile:force:quiet": "npm run compile:force -- --quiet",
-    "test": "npm run compile:force:quiet && hardhat test test/__setup.spec.ts test/PegasysIncentivesController/*.spec.ts test/StakedPSYS/*.spec.ts test/StakedPSYSV3/*.spec.ts",
```
Comment (Potolski) [inline package.json:17]: why did you change the scope of the test command?

Comment (tinkler) [reply package.json:17]: I forked the master branch, but there is no folder named ‘StakedPSYSV3’; only ‘StakedPSYSV2’ exists.

File: tasks/deployments/deploy-StakedTokenDataProvider.ts (hunk 10 above)
Comment (tinkler) [reply tasks/deployments/deploy-StakedTokenDataProvider.ts:41]: To ensure that AggregatedStakedTokenV3 continues to function properly, I left it unmodified and created a new interface 
'IAggregatedStakeToken' to integrate it with the 'Interface'. IAggregatedStakeToken includes the methods of StakedPSYS, StakedPSYS2, and StakedTokenV2Rev3, but we only need StakedTokenV2Rev3.

File: contracts/misc/StakedTokenDataProvider.sol (hunk 11 above)
Comment (tinkler) [reply contracts/misc/StakedTokenDataProvider.sol]: To ensure that AggregatedStakedTokenV3 continues to function properly, I left it unmodified and created a new interface
'IAggregatedStakeToken' to integrate it with the 'Interface'. IAggregatedStakeToken includes the methods of StakedPSYS, StakedPSYS2, and StakedTokenV2Rev3, but we only need StakedTokenV2Rev3.

File: package.json (hunk 12 above)
Comment (Potolski) [reply package.json:17]: But you still removed  _test/StakedPSYS/*.spec.ts_ even though the folder exists in both branches. Previously there were 112 tests (if im not mistaken) Now there are only 85.

File: contracts/misc/StakedTokenDataProvider.sol (hunk 11 above)
Comment (Potolski) [reply contracts/misc/StakedTokenDataProvider.sol]: Still not clear to me why the change was needed. Did you get this from somewhere or was it your idea? From What I see the change from _AggregatedStakedPSYSV3_ to _AggregatedStakedTokenV3_ removes some GHO's methods that might be useful since we'll release that token together with everything else. 

I just need to understand these changes so maintenance is easier later on.

Comment (tinkler) [reply contracts/misc/StakedTokenDataProvider.sol]: I left the implementation of AggregatedStakedPSYSV3 for the future when we release the GHO token. Currently, the IAggregatedStakeToken is being used by the interface. The code for StakedTokenDataProvider was obtained from etherscan, and I made some additions while integrating it with the 'Interface'.
The aave-stake-v2 that we forked lacks some code for the 'Interface'. I can't make it work with GHO now. Perhaps I didn't fully understand, but it seems that the 'Interface' is missing some code related to the GHO token.

File: package.json (hunk 12 above)
Comment (tinkler) [reply package.json:17]: I thought that the v1 version of StakedPSYS was no longer in use, so I took the liberty to remove the corresponding tests. I apologize for that, and I have now added them back.

File: contracts/misc/StakedTokenDataProvider.sol (hunk 11 above)
Comment (Potolski) [reply contracts/misc/StakedTokenDataProvider.sol]: 1. By interface you mean the interface repo right? (the frontend). 
2. The code from StakedTokenDataProvider was already on the repo but lacked the removal of BPT and other unused stuff (these changes I understood). What I still don't understand is why you changed the imports.
3. Why do you think there is something missing related to GHO on the interface? (I'm asking bc I haven't looked at that part of the code yet, so this will help me later on :) ) 
4. Is this the code you found on etherscan? https://etherscan.io/address/0x5e045cfb738f01bc73ceaff783f4c16e8b14090b#code

Comment (tinkler) [reply contracts/misc/StakedTokenDataProvider.sol]: 1. Yes, but the lending interface in org pegasys-fi is behind by 493 commits compared to aave/interface. They are using the StakingServiceV3, which handles the UI Data of GHO token.
2. I think I can take some time to make fewer changes, including not changing the imports.
3. The code of StakingServiceV3 is in @aave/contract-helpers@v1.28.0.
4. The URL of the code is https://etherscan.io/address/0xb12e82DF057BF16ecFa89D7D089dc7E5C1Dc057B#code.
